backends/p4tools/common/lib/symbolic_env.cpp

```diff
@@ -28,7 +27,7 @@ const IR::Expression *SymbolicEnv::get(const IR::StateVariable &var) const {
 bool SymbolicEnv::exists(const IR::StateVariable &var) const { return map.find(var) != map.end(); }
 
 void SymbolicEnv::set(const IR::StateVariable &var, const IR::Expression *value) {
-    map[var] = P4::optimizeExpression(value);
```

Comment:
If we remove the optimization, I'm not sure if the check if type is set should be here too? May not be necessary for all tools that use `SymbolicEnv`.